qiita_pet/handlers/study_handlers/sample_template.py

```diff
@@ -33,12 +33,22 @@ def get(self):
         # Get the most recent version for download and build the link
         download = sample_template_filepaths_get_req(
             study_id, self.current_user.id)
-        if 'filepaths' in download:
+
+        if download['status'] == 'success':
             download = download['filepaths'][-1]
+            dl_path = download_link_or_path(
+                is_local, download[1], download[0],
+                "Download sample information")
+        else:
+            dl_path = 'No sample information added'
+
+        stats = sample_template_summary_get_req(study_id, self.current_user.id)
```

Comment:
I think you need to remove the code in lines 53 and 54? Given that they're overwriting what you are doing in here.
That makes me wonder how this code has been tested.